producer/src/main/kotlin/streams/StreamsExtensionFactory.kt

```diff
@@ -69,13 +69,17 @@ class StreamsEventRouterLifecycle(val db: GraphDatabaseAPI, val streamHandler: S
 
     private fun unregisterTransactionEventHandler() {
         if (streamsEventRouterConfiguration.enabled) {
-            streamsConstraintsService.close()
+            if (::streamsConstraintsService.isInitialized) {
```

Comment:
this feels not right to have a static information
esp. in 4.0 with multi db this will fail as each db has it's own copy then
better to create an instance of: streamsConstraintsService
and have close be a no-op if the service is not initialized.

Reply:
perfect